src/core/server/saved_objects/service/lib/search_dsl/search_dsl.ts

```diff
@@ -65,16 +66,30 @@ export function getSearchDsl(
     throw Boom.notAcceptable('sortOrder requires a sortField');
   }
 
+  if (search && options.searchOptions) {
+    throw Boom.notAcceptable('`search` and `searchOptions` cannot be used together');
+  }
```

Comment:
Why only checking for `search` and not the other potentially conflicting options (`searchFields`, `rootSearchFields`, `defaultSearchOperator`)?